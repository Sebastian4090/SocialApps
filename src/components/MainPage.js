import React, { useState, useEffect } from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';

import './style/MainPage.css';


const MainPage = ({ socials, text , onSearchChange}) => {

    const [searchResults, setSearchResults] = useState([])

    useEffect(() => {
        let filteredSocials = socials.filter(social => {
            return(
                social.name.toLowerCase().includes(text.toLowerCase())
            );
        });
        setSearchResults(filteredSocials);
    }, [text, socials])

    const newSocial = searchResults;

    return (
            <div className='tc'>
                <Header />
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    <ErrorBoundry>
                        {
                            text === "" ? <CardList socials={socials}/> : <CardList socials={newSocial}/>
                        }
                    </ErrorBoundry>
                </Scroll>
            </div>
            );
        }
    
        

export default MainPage;