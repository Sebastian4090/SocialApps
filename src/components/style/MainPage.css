@font-face {
    font-display: swap;
    font-family: 'Rioma Regular';
    font-style: normal;
    font-weight: normal;
    src: local('Rioma Regular'), url('../../containers/Rioma-Regular-BF64dd839414854.woff') format('woff');
}

h1{
    font-family: 'Rioma Regular';
    font-weight: 200;
    color: lightblue;
}

img{
    width: 256px;
    height: 256px;
}