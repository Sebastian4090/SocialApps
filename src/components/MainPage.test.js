import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockSocials = [];
    const mockText = '';
    const mockOnSearchChange = jest.fn();

    wrapper = shallow(<MainPage socials={mockSocials} 
                                text={mockText} 
                                onSearchChange={mockOnSearchChange}
                                />);
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
});