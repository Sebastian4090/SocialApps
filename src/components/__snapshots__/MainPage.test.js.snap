// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Memo() />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundry>
      <CardList
        socials={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
